anki-editor-tests.el

```diff
@@ -168,5 +172,107 @@ Simple note body
+  "Test `anki-editor--note-at-point' should override subheading field."
+  (save-window-excursion
+    (with-current-buffer (anki-editor-test--test-org-buffer "test-files/property-fields.org")
+      (anki-editor-test--go-to-headline "\"Front\" property field with \"Front\" subheading")
+      (anki-editor-test--setup)
+      (unwind-protect
+          (should (equal
+                   (anki-editor-note-at-point)
+                   #s(anki-editor-note nil "Basic" "Tests"
+                                       (("Back" . "<p>\nShould be included\n</p>\n")
+                                        ("Front" . "<p>\nCan one define an anki-field inside an org-mode property?</p>\n"))
+                                       nil)))
+        (anki-editor-test--teardown)))))
+
+
+(ert-deftest test--note-at-point-for-note-with-unknown-property-field-should-raise-error ()
+  "Test `anki-editor--note-at-point' should raise error for unknown property field."
+  (save-window-excursion
+    (with-current-buffer (anki-editor-test--test-org-buffer "test-files/property-fields.org")
+      (anki-editor-test--go-to-headline "Foreign property field")
+      (anki-editor-test--setup)
+      (unwind-protect
+          (should
+           (equal
+            (should-error (anki-editor-note-at-point) :type 'user-error)
+            ' (user-error "Failed to map all named fields")))
+        (anki-editor-test--teardown)))))
+
+(ert-deftest test--note-at-point-for-examples-should-produce-correct-output ()
+  "Test `anki-editor--note-at-point' should produce correct output for examples."
+  (let ((test-items-alist
+         '(("Deck in file" . #s(anki-editor-note nil "Cloze" "Default"
+				                                 (("Back Extra" . "<p>\nDeck in file</p>\n")
+				                                  ("Text" . "<p>\nCards of this note will be created in {{c1::Default::which deck?}}\n</p>\n"))
+				                                 nil))
+           ("Deck in entry" . #s(anki-editor-note nil "Cloze" "Languages"
+                                 (("Back Extra" . "<p>\nDeck in entry</p>\n")
+                                  ("Text" . "<p>\nCards of this note will be created in {{c1::Languages::which deck?}}\n</p>\n"))
+                                 nil))
+           ("Raw fields" . #s(anki-editor-note nil "Basic" "Default"
+                                 (("Back" . "<p>\nWith property &lt;code&gt;:ANKI<sub>FORMAT</sub>: nil&lt;/code&gt;, content of the\nfield will be sent to Anki &lt;em&gt;unprocessed&lt;/em&gt;.  You can use\nwhatever Anki supports, like HTML tags.\n&lt;br&gt;\n&lt;br&gt;\nThis property is retrieved with inheritance, meaning that it can be\nset in any ancestor entries or at the top of the file with\n&lt;code&gt;#+PROPERTY: ANKI<sub>FORMAT</sub> nil&lt;/code&gt;, it's also possible to\noverride an outer level nil format with &lt;code&gt;:ANKI<sub>FORMAT</sub>: t&lt;/code&gt;.\n</p>\n")
```

Comment:
This looks unusual. I see these <code> blocks have been escaped. I think it's wrong.

Reply:
Actually, I'm wrong.